### PR TITLE
Adding SKE conditions during offer (provider)

### PR DESCRIPTION
**Pull request by elliotcm**
## Context

New feature allowing providers to specify SKE conditions as part of an offer, including special behaviour for language courses.

## Changes proposed in this pull request

Adds additional steps to the "make offer" flow for providers.

## Guidance to review

Test making an offer both with a modern language course and a non-language course.

## Things to check

- [x] This code does not rely on migrations in the same Pull Request

### CODE REVIEW

**Comment by elliotcm** [comment]
[⛷️ SKE: Provider - standard offer journey | Check details component & functionality](https://trello.com/c/YFbzEanG/1042-%E2%9B%B7%EF%B8%8F-ske-provider-standard-offer-journey-check-details-component-functionality)

**Comment by elliotcm** [comment]
[⛷️ SKE: Provider – eligibility criteria to show SKE journey](https://trello.com/c/KuOvHdB7/1086-%E2%9B%B7%EF%B8%8F-ske-provider-eligibility-criteria-to-show-ske-journey)

**Comment by carlosmartinez** [comment]
> **Database-level enum changes detected**
> 
> Please include a data migration for these attributes and values:
> 
> ```
> 36a37
> > <enum> SkeCondition.status: ["pending", "met", "unmet"]
> ```

Am I right in thinking that these are _not_ in fact database-level changes?

File: app/models/ske_condition.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class SkeCondition < OfferCondition
+  detail :language
+  detail :length
+  detail :reason
+
+  VALID_LANGUAGES = [
+    'French',
+    'Spanish',
+    'German',
+    'ancient languages',
```
Comment (carlosmartinez) [inline app/models/ske_condition.rb:10]: is this the right capitalisation?

**Comment by elliotcm** [comment]
@carlosmartinez 

> > **Database-level enum changes detected**
> > Please include a data migration for these attributes and values:
> > ```
> > 36a37
> > > <enum> SkeCondition.status: ["pending", "met", "unmet"]
> > ```
> 
> Am I right in thinking that these are _not_ in fact database-level changes?

Yeah I was going to look at that but the check is passing now.  I'm assuming it's an STI thing that we would need to exclude somehow.

File: app/models/ske_condition.rb (hunk 1 above)
Comment (elliotcm) [reply app/models/ske_condition.rb:10]: I believe so.  Assume because the first three are proper nouns.

**Review by carlosmartinez — APPROVED** [review]

**Pull request merged**